templates/rhsm-katello-reconfigure.erb

```diff
@@ -106,13 +106,19 @@ fi
   service goferd status >/dev/null && \
   service goferd restart >/dev/null 2&>1
 
+[ -f /usr/lib/systemd/system/goferd.service ] && \
 [ -f /bin/systemctl ] && \
   systemctl try-restart goferd >/dev/null 2>&1
 
-
-FQDN=`hostname -f`
-if [ $? == "0" ] && [ "$FQDN" != "localhost" ] && [ -d /etc/rhsm/facts/ ]; then
-  echo "{\"network.hostname-override\":\"$FQDN\"}" > /etc/rhsm/facts/katello.facts
+# EL5 systems and subscription-manager versions before 1.18.1-1 don't have the network.fqdn fact.
+# For these cases, we have to update the "hostname-override" fact
+if (test -f /etc/redhat-release && grep -q -i "Red Hat Enterprise Linux Server release 5" /etc/redhat-release) || \
+   (test -f /etc/centos-release && grep -q -i "CentOS Linux release 5" /etc/centos-release) || \
+   test ${RHSM_VERSION[0]:-0} -lt 1 -o ${RHSM_VERSION[1]:-0} -lt 18 -o \( ${RHSM_VERSION[1]:-0} -eq 18 -a ${RHSM_VERSION[2]:-0} -lt 2 \); then
+  FQDN=`hostname -f`
```

Comment:
@jlsherrill @johnpmitsch dont we want to still handle ```hostname -f``` erroring out ?

Reply:
The issue is with `set -e`, `hostname -f` failing will always cause the script to error out, even with `2> /dev/null` or as part of a conditional. I tried multiple ways wihout luck and came to the conclusion that the best way to handle it is to avoid running it completely if its not necessary and that avoids the vast majority of the occurrences since most systems are running a newer subman version

Reply:
or ```FQDN=$(hostname -f || :)``` ?
but then the check ```[ $? == "0" ]``` is useless and can be omitted

Reply:
@lpramuk I tried using hostname -f in a conditional as I mentioned above, it is still evaluated and fails the script